Source/Core/DolphinQt/CMakeLists.txt

```diff
@@ -1,15 +1,24 @@
-if (NOT Qt5_DIR AND MSVC)
+if (Qt6_DIR)
+	set(Qt_DIR "${Qt6_DIR}")
+endif()
+
+if (Qt5_DIR)
+	set(Qt_DIR "${Qt5_DIR}")
+endif()
+
+if (NOT Qt_DIR AND MSVC)
   if(_M_ARM_64)
-    set(Qt5_DIR "${CMAKE_SOURCE_DIR}/Externals/Qt/Qt5.15.0/msvc2019_arm64/lib/cmake/Qt5")
+    set(Qt_DIR "${CMAKE_SOURCE_DIR}/Externals/Qt/Qt5.15.0/msvc2019_arm64/lib/cmake/Qt5")
   else()
-    set(Qt5_DIR "${CMAKE_SOURCE_DIR}/Externals/Qt/Qt5.15.0/msvc2019_64/lib/cmake/Qt5")
+    set(Qt_DIR "${CMAKE_SOURCE_DIR}/Externals/Qt/Qt5.15.0/msvc2019_64/lib/cmake/Qt5")
   endif()
 endif()
 
-find_package(Qt5 5.9 REQUIRED COMPONENTS Gui Widgets)
+find_package(Qt NAMES Qt6 Qt5 COMPONENTS Core Gui Widgets REQUIRED)
+find_package(Qt${Qt_VERSION_MAJOR} 5.9 COMPONENTS Core Gui Widgets REQUIRED)
```

Comment:
Are those nested identifiers "the way to go" or can you just alias `Qt5`/`Qt6` to `Qt` (just like we do with `Qt_DIR`) and use things like `Qt::Core`, `Qt::Widgets` etc. instead?

Reply:
That would be cool, sadly
> We then do load the package for the determined Qt version again by creating the name Qt${QT_VERSION_MAJOR} on the fly. This is needed because CMAKE_AUTOMOC expects the package name to be either Qt5 or Qt6, and will print an error otherwise.

https://doc-snapshots.qt.io/qt6-dev/cmake-qt5-and-qt6-compatibility.html#supporting-older-qt-5-versions

I assume this extends to using ``Qt`` instead of ``Qt${Qt_VERSION_NUMBER}`` since both the Qt docs and Qt creator both use the ``${Qt_VERSION_NUMBER}`` approaches. It's probably best to not stray from that path or that might risk some breakages, even if it works right now.

Versionless Qt modules [will be possible](https://doc-snapshots.qt.io/qt6-dev/cmake-qt5-and-qt6-compatibility.html#versionless-targets) once we restrict support to Qt 5.15+ though.